### PR TITLE
chore: Add query tests for compacted chunks

### PR DESCRIPTION
**Pull request by alamb**
Re https://github.com/influxdata/influxdb_iox/pull/1855

# Rationale:
The query_tests scenarios do not currently include chunks created via "compaction" (they only test chunks created by load, rollover, etc); As I work to disable over eager deduplication for chunks created in the read buffer in  https://github.com/influxdata/influxdb_iox/pull/1855 and https://github.com/influxdata/influxdb_iox/issues/1763 I want to ensure sufficient coverage.

# Changes
1. Add a function `Db::compact_partition` to trigger a compaction
2. Adds a 7th scenario to two chunk secnarios where the data is compacted into a single large RUB chunk

# Testing FTW
This test has also uncovered a bug in `column_values` implementation of the read buffer, which I have filed as https://github.com/influxdata/influxdb_iox/issues/1860

### CODE REVIEW

File: query_tests/src/influxrpc/tag_values.rs [hunk 1]
```diff
@@ -212,8 +212,15 @@ async fn list_tag_values_table_and_timestamp_and_state_pred_state_col_no_rows()
...
     let expected_tag_keys = vec![];
+
+    // Ignore the compact case, until this is fixed:
+    // https://github.com/influxdata/influxdb_iox/issues/1860
+    let filter =
+        |setup: &DbScenario| setup.scenario_name != "Data in one compacted read buffer chunk";
+
     run_tag_values_test_case!(
-        TwoMeasurementsManyNulls {},
+        //TwoMeasurementsManyNulls {},
```
Comment (NGA-TRAN) [inline query_tests/src/influxrpc/tag_values.rs:222]: leftover?

File: query_tests/src/scenarios.rs [hunk 2]
```diff
@@ -823,8 +823,32 @@ pub async fn make_two_chunk_scenarios(
...
+            .await
+            .unwrap();
+    }
+    let table_names = write_lp(&db, data2).await; // write to MUB
+    for table_name in &table_names {
+        // compact chunks into a single RUB chunk
+        db.compact_partition(&table_name, partition_key)
+            .await
+            .unwrap();
+    }
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios.rs:844]: yay!

File: query_tests/src/scenarios.rs [hunk 3]
```diff
@@ -885,3 +909,42 @@ pub(crate) async fn make_one_rub_or_parquet_chunk_scenario(
...
+{
+    async fn make(&self) -> Vec<DbScenario> {
+        self.inner
+            .make()
+            .await
+            .into_iter()
+            .filter(|s| (self.filter)(s))
+            .collect()
+    }
+}
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios.rs:950]: I like thus filter infrastructure

File: server/src/db.rs [hunk 4]
```diff
@@ -1281,6 +1320,51 @@ mod tests {
...
+        let expected = vec![
+            "+-----+-------------------------------+",
+            "| bar | time                          |",
+            "+-----+-------------------------------+",
+            "| 1   | 1970-01-01 00:00:00.000000010 |",
+            "| 2   | 1970-01-01 00:00:00.000000020 |",
+            "+-----+-------------------------------+",
+        ];
+        let batches = run_query(Arc::clone(&db), "select * from cpu").await;
+        assert_batches_eq!(&expected, &batches);
```
Comment (NGA-TRAN) [inline server/src/db.rs:1365]: nice

**Review by NGA-TRAN — APPROVED** [review]
LGTM

File: query_tests/src/influxrpc/tag_values.rs (hunk 1 above)
Comment (alamb) [reply query_tests/src/influxrpc/tag_values.rs:222]: I was trying to hint that once the bug was fixed, we could just incomment that line 🤔  though now that I type that it is not as clear to me it is clearer

File: server/src/db.rs [hunk 5]
```diff
@@ -370,6 +370,45 @@ impl Db {
...
+    /// Compacts all chunks in a partition to create a new chunk
+    ///
+    /// This code does not do any checking of the read buffer against
+    /// memory limits, etc
+    ///
+    /// This (async) function returns when this process is complete,
+    /// but the process may take a long time
+    ///
+    /// Returns a handle to the newly created chunk in the read buffer
+    pub async fn compact_partition(
```
Comment (alamb) [inline server/src/db.rs:382]: FYI @tustvold

**Pull request merged**